simuvex/plugins/posix.py

```diff
@@ -141,6 +141,15 @@ def set_state(self, state):
             if self.state is not f.state:
                 raise SimError("states somehow differ")
 
+        # Do this for the filesystem also
+        for fd, f in self.fs.iteritems():
```

Comment:
In order to access the contents of the files stored in the fs, we need to keep the states synchronized. This seems to work for me.